www/codeigniter/application/controllers/master/Stock.php

```diff
@@ -139,16 +139,15 @@ private function create_empty_material()
 			'color_number_code'=> NULL,
 			'color_number_tint'=> NULL,
 			'spec'=> NULL,
+			'unit'=> NULL,
 			'display_type'=> NULL,
 			'send_order_customer_id'=> NULL
 		];
 	}
 
 	private function validate_form()
 	{
-		$this->form_validation->set_rules('part_number', '品番', 'trim|required|regex_match[/^[a-zA-Z0-9_\-]+$/]');
-		$this->form_validation->set_rules('unit', '単位', 'trim|required|regex_match[/^[a-zA-Z0-9_\-]+$/]');
-		$this->form_validation->set_rules('send_order_customer_id', '発注先', 'trim|required|regex_match[/^[a-zA-Z0-9_\-]+$/]');
+		$this->form_validation->set_rules($this->materials_model->get_validation());
 		if (!$this->form_validation->run())
 		{
 			$this->session->set_flashdata('error_message', validation_errors());
```

Comment:
@trandinhlam 
**index()**, load_pagination có URL nhiều tham số nên dùng http_build_query
**insert(), update() và delete()** kiểm tra thêm điều kiện 
if (!$this->is_post_request())
		{
			redirect('/master/account');
		}